msfvenom

```diff
@@ -49,10 +49,10 @@ def init_framework(create_opts={})
   ]
 
   create_opts[:module_types].map! do |type|
-    type = Msf.const_get("MODULE_#{type.upcase}")
+    Msf.const_get("MODULE_#{type.upcase}")
   end
 
-  @framework = ::Msf::Simple::Framework.create
+  @framework = ::Msf::Simple::Framework.create(create_opts)
```

Comment:
Maybe a dumb question - but am I missing a piece of the glue here? 🤔 Does this need to be passing in `DeferModuleLoads` to have the performance gains for the scenario of generating payloads?

For me when running:

```
time bundle exec ./msfvenom -p android/meterpreter/reverse_tcp LHOST=192.168.1.223 LPORT=4444
```

`create_opts` is sent through as:

```
> create_opts
=> {:module_types=>["payload", "encoder", "nop"]}
```

So if we're generating a single payload, it's still going to load quite slowly as `DeferModuleLoads` hasn't been set? 🤔

Reply:
From the reading the code a bit more - if I understand correctly this PR will fix a performance regression in something like `bundle exec ./msfvenom --list platforms` - as that specifies specifically to load module types as `[]`:

https://github.com/rapid7/metasploit-framework/blob/177f720f800f215dee66c33d33d3d99e4096ddbd/msfvenom#L251-L254

But for the main scenario of using venom to generate payloads - although we're now correctly skipping the load of aux modules etc by fixing this regression, it looks like we might be able to do more to reduce the payload generation time further? 🕵️

Reply:
We can for sure do more to reduce the time, we're still spending ~2 seconds on loading other payloads that aren't being used but I think it's still worth landing this as is first

Reply:
> on loading other payloads that aren't being used

Just confirming that this is meant to be module sets rather than payloads? 👀 

Landing this now; I think we'll be able to shave off more time from fixing the autoloading issues - such as eagerly loading the faker gem, and deferring the calculation of dynamic payload sizes. We talked about loading only the payload required too, but that might be a can of worms. Onwards for now! 📈